### ISSUE TITLE
Bug: threaded scheduler is not interruptible

### ISSUE THREAD

**Issue by abast** [dask/dask#2143]
```
import dask, time
dask.__version__
> u'0.14.1'

# can be interrupted
fun = dask.delayed(lambda x: time.sleep(5))
dask.compute([fun(x) for x in range(100)], get = dask.threaded.get)

# can't be interrupted
fun2 = dask.delayed(lambda x: sum([1 for x in range(10000000)]))
dask.compute([fun2(x) for x in range(100)], get = dask.threaded.get)
```

Using the threaded scheduler, I noticed that some tasks cannot be interrupted by pressing Ctrl+C in a python console. Directly after pressing Ctrl+C the CPU usage of that process jumps up: It uses all available minus one cores. Judging by the output of top, the process seems to spend the computing time in the kernel space.

**Comment by jcrist** [comment]
Thanks for the bug report. This is due to https://bugs.python.org/issue1360, and is fixed in python 3. A workaround is to always specify a timeout to `get` (can be extremely long). Fix coming soon. Minimal example:

```python
from time import sleep
from multiprocessing.pool import ThreadPool
from dask.compatibility import Queue

p = ThreadPool()
queue = Queue()
p.apply_async(sleep, args=(10,), callback=queue.put)
queue.get()   # This will block KeyboardInterrupt in python 2.7
# queue.get(1000000)  # This won't
```

**Comment by jcrist** [comment]
Fixed by #2144.

**Comment by abast** [comment]
As usual I am really amazed by your speed. Thanks!
